tools/sloop.c

```diff
@@ -0,0 +1,84 @@
+/**
+ * Library of functions pertaining to Sloop shadows. Slooping is a way
+ * to simulate the built-in snooping functionality of EotL.
+ *
+ * @author gon
+ * 2017-05
+ */
+
+inherit AnsiTellCode;
+
+#include <ansi.h>
+
+#define SLOOPER_SHADOW "/usr/gon/tools/slooper_shad"
+#define SLOOPEE_SHADOW "/usr/gon/tools/sloopee_shad"
+
+/**
+ * Function Declarations
+ */
+void add_sloopee_shadow( object ob );
+void add_slooper_shadow( object ob );
+void dest_sloopee_shadow( object ob );
+void dest_slooper_shadow( object ob );
+object * find_sloopee_shadow( object ob );
+object * find_slooper_shadow( object ob );
+
+/**
+ * When function is called, adds SLOOPEE_SHADOW to object ob.
+ * @param  object ob object to add SLOOPEE_SHADOW
+ */
+void add_sloopee_shadow( object ob ) {
+    object shadow = clone_object( SLOOPEE_SHADOW );
+    shadow->sh_init( ob );
+}
+
+/**
+ * When function is called, adds SLOOPER_SHADOW to object ob.
+ * @param  object ob object to add SLOOPER_SHADOW
+ */
+void add_slooper_shadow( object ob ) {
+    object shadow = clone_object( SLOOPER_SHADOW );
+    shadow->sh_init( ob );
+}
+
+/**
+ * When function is called, uses find_sloopee_shadow() and if that
+ * returns ! 0, dests the shadow object returned from the array.
+ * @param  object ob            object that holds the shadow if find_sloopee_shadow returns non zero.
+ */
+void dest_sloopee_shadow( object ob ) {
+    object * arr = find_sloopee_shadow( ob );
+    remove_shadow( arr[0] );
```

Comment:
Wow. I had no idea, but now that I understand this, it makes perfect sense. I'd like to know more about this next time we talk. Are there rules to clean up abandoned but not destructed objects? How does this impact memory usage?